api/src/test/java/uk/gov/dhsc/htbhf/eligibility/EligibilityServiceIntegrationTests.java

```diff
@@ -0,0 +1,239 @@
+import static uk.gov.dhsc.htbhf.eligibility.testhelper.HMRCEligibilityRequestTestDataFactory.anHMRCEligibilityRequestWithEligibilityDates;
+import static uk.gov.dhsc.htbhf.eligibility.testhelper.HMRCEligibilityResponseTestDataFactory.anHMRCEligibilityResponseWithStatus;
+import static uk.gov.dhsc.htbhf.eligibility.testhelper.PersonDTOTestDataFactory.aPerson;
+import static uk.gov.dhsc.htbhf.eligibility.testhelper.TestConstants.SIMPSON_DWP_HOUSEHOLD_IDENTIFIER;
+import static uk.gov.dhsc.htbhf.eligibility.testhelper.TestConstants.SIMPSON_HMRC_HOUSEHOLD_IDENTIFIER;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class EligibilityServiceIntegrationTests {
+
+    private static final URI ENDPOINT = URI.create("/v1/eligibility");
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @MockBean
+    private RestTemplate restTemplateWithIdHeaders;
+
+    @Value("${hmrc.base-uri}")
+    private String hmrcUri;
+
+    @Value("${dwp.base-uri}")
+    private String dwpUri;
+
+    @Value("${eligibility-check-period-length}")
+    private Integer eligibilityCheckPeriodLength;
+
+    @Test
+    void shouldReturnEligibleResponseGivenEligibleResponseReturnedFromBothDwpAndHmrc() {
+        runIntegrationTestReturningEligibilityResponse(ELIGIBLE, ELIGIBLE, anEligibleEligibilityResponse());
+    }
+
+    @ParameterizedTest(name = "Should return ELIGIBLE response from eligibility-service when DWP status returned is ELIGIBLE and HMRC is {0}")
+    @ValueSource(strings = {
+            "INELIGIBLE",
+            "PENDING",
+            "ERROR",
+            "NO_MATCH"
+    })
+    void shouldReturnEligibleResponseGivenEligibleResponseFromDwp(EligibilityStatus hmrcResponseEligibilityStatus) {
+        runIntegrationTestReturningEligibilityResponse(ELIGIBLE,
+                hmrcResponseEligibilityStatus,
+                anEligibilityResponseWithDwpHouseholdIdentifier(ELIGIBLE, SIMPSON_DWP_HOUSEHOLD_IDENTIFIER));
+    }
+
+    @ParameterizedTest(name = "Should return {2} response from eligibility-service when DWP status returned is {0} and HMRC is {1}")
+    @CsvSource({
+            "PENDING, PENDING, PENDING",
+            "PENDING, INELIGIBLE, PENDING",
+            "PENDING, ERROR, PENDING",
+            "PENDING, NO_MATCH, PENDING",
+            "ERROR, ERROR, ERROR",
+            "ERROR, INELIGIBLE, ERROR",
+            "ERROR, PENDING, PENDING",
+            "ERROR, NO_MATCH, ERROR",
+            "INELIGIBLE, PENDING, PENDING",
+            "INELIGIBLE, INELIGIBLE, INELIGIBLE",
+            "INELIGIBLE, NO_MATCH, INELIGIBLE",
+            "INELIGIBLE, ERROR, ERROR",
+            "NO_MATCH, INELIGIBLE, INELIGIBLE",
+            "NO_MATCH, PENDING, PENDING",
+            "NO_MATCH, ERROR, ERROR"
+    })
+    void shouldReturnEligibilityResponseGivenANonEligibleResponseFromEitherDwpOrHmrc(EligibilityStatus dwpResponseEligibilityStatus,
```

Comment:
do we need an integration test for each combination? We're covering all of them in unit tests  - I'd suggest that just a couple of combinations to sufficient. We could probably do without this test altogether, and cut down the number of combinations in the dwp- and hmrc-specific methods. Thoughts?...

Reply:
Yes, I did wonder about that, but I thought the different permutations were crucial to how the service works and adding the extra permutations wasn't that difficult to do for little outlay. Yes, I agree they're unit tested, but any subset I picked would seem quite arbitrary. I do think `NO_MATCH` needs to be here though as it returns a different HTTP status code.

Reply:
The need to NO_MATCH tests is a good point. How about ELIGIBLE and NO_MATCH as the subset then?